### PR TITLE
feat(hot): adds vote check to hot repos

### LINKED ISSUE #230: Feature: Check if the user has voted on the repo
### Type of feature

🍕 Feature

### Current behavior

This https://github.com/open-sauced/hot/pull/216 introduces the new UI to show if the user has voted. Currently the app does not have a function to check votes based on repo and user id. 

### Suggested solution

Create a function that checks the API if the user has already voted on the repo on load. this will eventually need to be cached in a global state store to prevent multiple checks on refresh or pagination. 

### Additional context
Screen Shot 2022-07-25 at 9 02 41 AM

Note the vote button states "voted." This interaction is not possible to check with this new function.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by bdougie**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

Working this voted/upvote button. Turns out that bun is not on the repo list and is unable to be voted on. We should only highly repos that exist in the list.

## Related Tickets & Documents

fixes #230
also #246 

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
I don't think the fetched data is filter votes as currently implemented.

**Comment by 0-vortex** [comment]
I've added the top 3 hot repos to the repo and contributions processing cron jobs - they should be available in a max of 6 hours and voting should work, regardless of static array.

Changing the ids of the top 3 repos to something we have in the database works, however there is a bug removing all the repos in the list repository when clicking on my votes. I'm investigating that bug, but essentially the PR is ready for review, it's just the backend that's a bit behind here! 🍕

**Comment by bdougie** [comment]
Confirmed I can vote and it reflects that in the ui

**Comment by bdougie** [comment]
@open-sauced/triage this one is ready for you to review.

File: src/components/HotRepositories.tsx [hunk 1]
```diff
@@ -92,6 +94,17 @@ const HotRepositories = ({ user }: HotReposProps): JSX.Element => {
     setHotRepos([...hotRepos]);
   };
 
+  useEffect(() => {
+    fetchRecommendations("myVotes", 1000, user, "").then((data) => {
+      const flatArrayIds = data.map((repo) => repo.id);
+      // hotRepos.forEach((repo) => {
```
Comment (0-vortex) [inline src/components/HotRepositories.tsx:100]: this commented code should be put back in IMHO

Comment (bdougie) [reply src/components/HotRepositories.tsx:100]: Yeah forgot to push that. Doing it now

**Comment by CMCSWEENE** [comment]
Upvoted changes to voted when signed in with GitHub. Glitches slightly at first but after hitting upvoted multiple times it works. #252

**Comment by 0-vortex** [comment]
> Upvoted changes to voted when signed in with GitHub. Glitches slightly at first but after hitting upvoted multiple times it works. #252 

Good catch, feel free to add a new issue for this! 🍕

**Comment by jsmith2320** [comment]
Everything looks good

**Pull request merged**